.github/workflows/pypi_build_and_images.yaml

```diff
@@ -40,18 +40,62 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: "3.8"
+
+      # setup ubuntu Linux with packages required for installing locally WM
+      # python packages, e.g. we need curl-config for pycurl (it comes from dev
+      # libcurl library), etc.
+      - name: Update system packages to include pycurl
+        run: |
+          echo "update system packages to include pycurl dev librarty (required for pycurl)"
+          sudo apt update
+          sudo apt install -y curl libcurl4-openssl-dev
+
+      # upgrade pip
       - name: Upgrade pip3
         run: |
           python3 -m pip install --upgrade pip
+
+      # create appropriate setup.py for python builds
       - name: Update the setup script template with package name
         run: |
           sed "s/PACKAGE_TO_BUILD/${{ matrix.target }}/" setup_template.py > setup.py
+
+      # perform CI/CD actions to install WM packages locally
+
+      # first we create proper requirements file, but will skip gfal dependency
+      # as it only exist for CERN/WLCG based Linux distribution
       - name: Create requirements file
         run: |
-          cp requirements.txt requirements.wmcore.txt
-          awk "/(${{ matrix.target }}$)|(${{ matrix.target }},)/ {print \$1}" requirements.wmcore.txt > requirements.txt
-      - name: Build sdist
-        run: python3 setup.py clean sdist
+          echo "create new requirements.txt without gfal dependencies for CI/CD"
+          cat requirements.txt | egrep -v "gfal" > requirements.${{ matrix.target }}.txt
```

Comment:
If we are going to skip gfal, then it is better to use the full name of the package here to avoid any potential future mistake.
Nonetheless, if we skip `gfal2-python` python library here, I am almost sure that this will break MSUnmerged.

Either we test this out before merging these changes; or we merge it, and you test a new release candidate for MSUnmerged to. confirm that everything is Okay.

Reply:
Yes, I think you are correct, Dario. Looking again at these lines:
https://github.com/dmwm/CMSKubernetes/blob/master/docker/pypi/reqmgr2ms-unmerged/Dockerfile#L14-L17

we install packages listed in master/requirements.txt - but we skip gfal2. Then when we install MSUnmerged for PyPi, we are also saying that we don't want to install any of its dependencies.

I guess the Python `gfal2-python` package is equivalent to the Conda `python-gfal2`(?) If so, I see no need to change any of that in this ticket and we should be good then.

Reply:
ok, I used `cat requirements.txt | egrep -v "gfal2-python|python-gfal2"` to satisfies both cases.